File: go/bottle-song/bottle_song.go
package bottlesong

import (
	"fmt"
	"strings"
)

var numbers = []string{"No", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten"}

func bottles(num int) string {
	var format string
	if num == 1 {
		format = "%s green bottle"
	} else {
		format = "%s green bottles"
	}
	return fmt.Sprintf(format, numbers[num])
}

// Recite returns song verses.
func Recite(startBottles, takeDown int) []string {
	verses := make([]string, 0, takeDown*5)
	for i := startBottles; i > startBottles-takeDown; i-- {
		lines := []string {
			fmt.Sprintf("%s hanging on the wall,", bottles(i)),
			fmt.Sprintf("%s hanging on the wall,", bottles(i)),
			"And if one green bottle should accidentally fall,",
			fmt.Sprintf("There'll be %s hanging on the wall.", strings.ToLower(bottles(i-1))),
			"",
		}

		verses = append(verses, lines...)
	}
	return verses[:takeDown*5-1]
}

File: python/bottle-song/bottle_song.py
"""Bottle Song."""
NUMBERS = "No One Two Three Four Five Six Seven Eight Nine Ten".split()


def bottles(num: int) -> str:
    """Return the bottle term."""
    out = f"{NUMBERS[num]} green bottle"
    if num != 1:
        out += "s"
    return out


def recite(start: int, take: int = 1) -> list[str]:
    """Return verses."""
    out = []
    for i in range(start, start - take, -1):
        out.append(f"{bottles(i)} hanging on the wall,")
        out.append(f"{bottles(i)} hanging on the wall,")
        out.append("And if one green bottle should accidentally fall,")
        out.append(f"There'll be {bottles(i - 1).lower()} hanging on the wall.")
        out.append("")
    return out[:-1]
